scm-record/src/types.rs

```diff
@@ -74,71 +74,54 @@ pub enum RecordError {
     Bug(String),
 }
 
-/// The Unix file mode. The special mode `0` indicates that the file did not exist.
+/// The file mode.
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct FileMode(pub usize);
+pub enum FileMode {
+    /// A Read Write Execute style Unix file mode
+    Rwx(usize),
+
+    /// Indicates that the file did not exists.
+    Absent,
+}
 
 impl FileMode {
-    /// Get the file mode corresponding to an absent file. This typically
-    /// indicates that the file was created (if the "before" mode) or deleted
-    /// (if the "after" mode).
-    pub fn absent() -> Self {
-        Self(0)
-    }
+    /// The default Unix permissions for files.
+    pub(crate) const FILE_DEFAULT: FileMode = FileMode::Rwx(0o100644);
 }
 
 impl Display for FileMode {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let Self(mode) = self;
-        write!(f, "{mode:o}")
+        match self {
+            FileMode::Rwx(mode) => {
+                write!(f, "{mode:o}")
+            }
+            FileMode::Absent => {
+                write!(f, "<absent>")
+            }
+        }
     }
 }
 
 impl From<usize> for FileMode {
     fn from(value: usize) -> Self {
-        Self(value)
-    }
-}
-
-impl From<FileMode> for usize {
```

Comment:
These didn't seem to be being used, either here, in the tests, or in `jj`. Now that `0` isn't a sentinel value there isn't an infallible conversion to `FileMode` anyway, so I've removed it. But if I'm missing something, I can add these back in.